README.md

```diff
@@ -138,9 +138,11 @@ Kube-secret-syncer maintains both the list of AWS Secrets as well as their value
 
 ## [Security model](#security-model)
 
-By default, kube-secret-syncer will use the Kubernetes node's IAM role to list and retrieve the secrets. However, when
-synced secrets have an IAMRole field defined, kube-secret-syncer will assume that role before retrieving the secret. This
-implies that the role specified by IAMRole can be assumed by the role of the Kubernetes node kube-secret-syncer runs on.
+Kube-secret-syncer relies on the AWS Go SDK to communicate with AWS - and supports the different ways of
+authenticating to AWS described in the [AWS Go SDK documentation](https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials).
```

Comment:
I think we should explicitly point out that additional steps are required to prevent that the iam roles used kube-secret-syncer aren't assumable by other pods in the k8s cluster as this would otherwise break the security model.

E.g. You shouldn't use ec2 instance profile based authentication and you need to enable namespace restrictions if you use kube2iam https://github.com/jtblin/kube2iam#namespace-restrictions, which isn't enabled by default.

Reply:
I'd second this -- at a minimum it's probably worth calling out which auth methods would break the security model, and the necessary requirements to maintain the security model, like:

> prevent that the iam roles used kube-secret-syncer aren't assumable by other pods in the k8s cluster

I can understand not giving explicit setup instructions (for kube2iam, kiam, IRSA, etc.) as the projects themselves will always have the most complete/up-to-date documentation. But might also be worth a link or two to some of those projects that can be configured to meet the requirements?